tests/unit/multi_select_test.coffee

```diff
@@ -0,0 +1,96 @@
+multiSelect = null
+dispatcher = null
+
+module "[Unit] Multi-Select Component unit tests",
+  setup: ->
+    Ember.run ->
+      dispatcher = Ember.EventDispatcher.create()
+      dispatcher.setup()
+      multiSelect = Ember.Widgets.MultiSelectComponent.create
+        content: ['foo', 'bar', 'barca', 'baz']
+
+  teardown: ->
+    Ember.run ->
+      dispatcher.destroy()
+      multiSelect.destroy()
+
+test 'Test preparedContent after some options are already selected', ->
+  expect 2
+  multiSelect.set 'selections', ['bar', 'baz']
+  multiSelect.set 'query', 'ba'
+  equal(multiSelect.get('preparedContent').length, 1)
+  equal(multiSelect.get('preparedContent')[0], 'barca')
+
+test 'Test keyboard and mouse interaction', ->
+  expect 17
+  selectedText = null
+
+  Ember.run ->
+    append(multiSelect)
+
+  Ember.run ->
+    multiSelectComponent = multiSelect.$()
+    highlightedComponent = find '.ember-select-multi', multiSelectComponent
+    textField = find '.ember-text-field', multiSelectComponent
+
+    validateFocusAndDropdown = (messageVisible, messageFocus) ->
+      ok isVisible(find '.ember-select-results', multiSelectComponent, messageVisible)
+      ok isFocus(textField,multiSelectComponent, messageFocus)
+
+    multiSelectComponent.focus()
+    wait().then ->
+      # test pressing ENTER key to open dropdown
+      keyEvent(multiSelectComponent, 'keydown', 13)
+    .then ->
+      ok isVisible(find '.ember-select-results', multiSelectComponent, 'Dropdown list should appear after pressing Enter')
+      resultItems = find '.ember-select-result-item', multiSelectComponent
+      ok $(resultItems[0]).hasClass('highlighted'), 'The first option should be highlighted'
+      # test selecting option using ENTER key
+      selectedText = $(resultItems[0]).text().trim()
+
+      keyEvent(multiSelectComponent, 'keydown', 13)
```

Comment:
you should make a helper for these `keyEvent`s. Something like `pressEnter(multiSelectComponent)` would read better here.